packages/core/lib/aws-xray.js

```diff
@@ -3,16 +3,7 @@ var logging = require('./logger');
 var segmentUtils = require('./segments/segment_utils');
 var utils = require('./utils');
 var LambdaEnv = require('./env/aws_lambda');
-
-// Import Data from package.json,
-// If the importing of package.json fails leave
-// pkginfo as an empty object
-var pkginfo = {}
-try {
-  pkginfo = require('../package.json');
-} catch (err) {
-  logging.getLogger().debug('Failed to load SDK data:', err);
-}
+var pkginfo = module.filename ? require('../package.json') : {};
```

Comment:
@willarmiros this change is perhaps a little closer to the behavior that was in place originally. I believe the `module.filename` check is there for something like webpack which doesn't play well with pgkinfo (but in a less crash and burn on startup way then a bundler like rollup).
However the other thing I've been thinking about is if this change is safe as a point release. In the case of bundlers this change may break because importing a json file will usually require a json loader which may or may-not be configured. However if the json loader is present we can start getting the xray sdk data loaded when previously it would have been unknown. I'm happy to defer to you in terms of which change makes the most sense.

1) require package.json in a try/catch and log an error if it can't be required - Probably a result of bundling
2) skip the require call if modile.filename is falsey - probably turns off require call for some bundlers
3) Unconditional require (my preference) - AWS-SDK won't be usable if the require fails but the code is the cleanest and the failure cases involve running in non-node environments.

Reply:
Reading through [the source](https://github.com/indexzero/node-pkginfo/blob/master/lib/pkginfo.js), pkginfo is basically just requiring `../package.json` exactly as you are now. Though I saw their `require` is [wrapped by try/catch](https://github.com/indexzero/node-pkginfo/blob/master/lib/pkginfo.js#L103-L105), so I was wondering how this breaks in a non-standard environment as you said in #368?

Regardless, I would not be in favor of option 3, because even if we held this change til a major version bump making the SDK unusable for existing customers simply because they bundle their `package.json` unorthodoxly is silly. I'd be in favor of option 1, but at that point it seems like we're reinventing the wheel and just doing exactly the work of `pkginfo` ourselves.

Reply:
@willarmiros I'm happy to leave the version with the try/catch/log setup. I wrote up some notes on the failure case on the [original issue](https://github.com/aws/aws-xray-sdk-node/issues/368#issuecomment-769348677). I'll drop the other commits off this PR and build a test setup where I can see what will happen if a bundler is not setup to load json and this pr where to be applied. I'll add what I find to this issue.

As for replicating the work of pkginfo ourselves directly requiring the package.json from a known location is a pretty substantial improvement. The `find` code in pkginfo does several directory checks, directory traversals and tries to require a package.json file. From a performance standpoint this library will require substantially less file access/read calls with the direct import. Beyond that a properly configured bundler that can load json will now be able to correctly include the x-ray-sdk data in it's payloads.

Reply:
Ok I'm sold on the idea that this require statement is an improvement on using `pkginfo`.